### PR TITLE
Handle Missing Headers in Data Dictionary Upload

### PR DESCRIPTION
**Pull request by zandre-eng**
## Product Description

If there are any blank header cells that don't get loaded when processing a DD upload then the following error will be shown:
Screenshot from 2024-04-02 12-53-22

## Technical Summary

Link to ticket [here](https://dimagi.atlassian.net/browse/SC-3581).

An issue ([ticket](https://dimagi.atlassian.net/browse/QA-6279) for ref) was picked up where a Data Dictionary Excel upload resulted in an `IndexError` exception. Specifically, this occurred because the length of the `column_headings` list in `map_row_values_to_column_names()` was less than the length of a row that contained data. Normally, this shouldn't happen as `column_headings` should have `None` values for blank header cells.

This is a very specific error which I was unable to reproduce either locally or on the production environment. Leaving the header cells blank for columns that contained data resulted in the correct UI error being thrown. This seems to be a very special case where the Excel file got corrupted and so some of the blank header cells weren't being loaded with `worksheet.iter_rows()`.

This PR adds in an additional safety precaution by correctly catching the exception and raising a UI error for the user.

## Feature Flag

None

### Safety story

- Local testing
- Unit tests

This change will only impact domains that have the Data Dictionary privilege.

### Automated test coverage

Unit tests exist for both the `map_row_values_to_column_names()` and `get_column_headings()` util functions.

### QA Plan

No QA planned.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/data_dictionary/tests/test_util.py [hunk 1]
```diff
@@ -210,15 +210,21 @@ def test_get_column_headings(self):
...
+        )
         expected_output = {
             'col_1': 'a',
             'col_2': 'b',
             'col_3': 'empty',
         }
+        self.assertEqual(len(errors), 0)
         for key, val in expected_output.items():
             self.assertEqual(row_vals[key], val)
+        row_vals, errors = map_row_values_to_column_names(self.row, column_headings[0:1], sheet_name=sheet_name)
```
Comment (ajeety4) [inline corehq/apps/data_dictionary/tests/test_util.py:226]: nit: consider putting this test in a separate test method for better isolation of the failure scenario.

File: corehq/apps/data_dictionary/bulk.py [hunk 2]
```diff
@@ -103,7 +103,12 @@ def _process_multichoice_sheets(domain, workbook, allowed_value_info, prop_row_i
             if len(row) < 1:
                 continue
 
-            row_vals = map_row_values_to_column_names(row, column_headings, default_val='')
+            row_vals, row_val_errors = map_row_values_to_column_names(
```
Comment (ajeety4) [inline corehq/apps/data_dictionary/bulk.py:106]: I was wondering if the errors would be repeated for all the rows if a specific column header is absent ?

**Review by ajeety4 — COMMENTED** [review]
The changes looks good to me. Have only one non-blocking query.

File: corehq/apps/data_dictionary/bulk.py (hunk 2 above)
Comment (zandre-eng) [reply corehq/apps/data_dictionary/bulk.py:106]: This wouldn't. This is in a loop that processes one row at a time. If there are any `row_val_errors` for a row then the entire loop will be exited with the relevant errors.

File: corehq/apps/data_dictionary/tests/test_util.py (hunk 1 above)
Comment (zandre-eng) [reply corehq/apps/data_dictionary/tests/test_util.py:230]: [9bdcafa](https://github.com/dimagi/commcare-hq/pull/34368/commits/9bdcafacbfc4e5fa9f8b908d721d6b0a3426d287)

File: corehq/apps/data_dictionary/bulk.py (hunk 2 above)
Comment (ajeety4) [reply corehq/apps/data_dictionary/bulk.py:106]: Great. Thanks for the clarification.

**Review by ajeety4 — APPROVED** [review]

**Pull request merged**